rust/Cargo.toml

```diff
@@ -182,6 +182,7 @@ tracing-error = "0.2"
 tracing-futures = "0.2"
 tracing-subscriber = { version = "0.3", default-features = false }
 tracing-test = "0.2.2"
+typetag = "0.2"
```

Comment:
`Serialize` isn't object safe so we can't normally use it for our `Box<dyn PendingOperation>` types, but this crate works around that by creating a `Serialize`-able enum whose each item is the struct `impl` of the `PendingOperation` trait.